src/app/beneficiaries/beneficiaries-form/beneficiaries-form.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { BeneficiariesService } from '../beneficiaries.service';
+import { BeneficiariesForm } from './beneficiaries-form.model';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-beneficiaries-form',
+  templateUrl: './beneficiaries-form.component.html',
+  styleUrls: ['./beneficiaries-form.component.scss']
+})
+export class BeneficiariesFormComponent implements OnInit {
+
+  accountDetails: any;
+  accountTypeOptions: any;
+  editMode: boolean = false;
+  beneficiariesForm: FormGroup;
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private formBuilder: FormBuilder,
+              private beneficiariesService: BeneficiariesService,
+              private toastr: ToastrService) {
+    this.route.data.subscribe((data: {accountTypeOptions: any}) => {
+      const { accountTypeOptions } = data.accountTypeOptions;
+      this.accountTypeOptions = accountTypeOptions;
+    });
+    if (this.router.getCurrentNavigation().extras.state?.editMode){
+      this.accountDetails = this.router.getCurrentNavigation().extras.state;
+      this.editMode = true;
+    }
+  }
+
+  ngOnInit(): void {
+    this.createBeneficiaryForm();
+  }
+
+  createBeneficiaryForm() {
+    const transferLimit = this.accountDetails?.transferLimit;
+    this.beneficiariesForm = this.formBuilder.group({
+      name: [this.accountDetails?.name, Validators.required],
+      officeName: [{value: this.accountDetails?.officeName, disabled: this.editMode}, Validators.required],
+      accountNumber: [{value: this.accountDetails?.accountNumber, disabled: this.editMode}, Validators.required],
+      accountType: [{value: this.accountDetails?.accountType?.id, disabled: this.editMode}, Validators.required],
+      transferLimit: [transferLimit == 0? undefined: transferLimit, Validators.min(1)]
+    });
+  }
+
+  submitBeneficiaryForm() {
```

Comment:
@ksraj123  submitBeneficiaryForm() function has more than one exit point, let's refactor it to achieve cleaner code.